src/routes/safe/components/Transactions/TxList/TxQueueRow.tsx

```diff
@@ -22,22 +23,27 @@ type TxQueueRowProps = {
 
 export const TxQueueRow = ({ isGrouped = false, transaction, onChildExpand }: TxQueueRowProps): ReactElement => {
   const { activeHover } = useContext(TxHoverContext)
+  const { activeHover: batchTxActiveHover } = useContext(BatchExecuteHoverContext)
   const [tx, setTx] = useState<Transaction>(transaction)
   const willBeReplaced = tx.txStatus === LocalTransactionStatus.WILL_BE_REPLACED ? ' will-be-replaced' : ''
-  const isPending = useSelector((state: AppReduxState) => isTxPending(state, transaction.id))
-  const pendingTx = useSelector(pendingTxByChain)
-  const pendingTxNonce = isMultisigExecutionInfo(pendingTx?.executionInfo) ? pendingTx?.executionInfo.nonce : undefined
-  const nonce = isMultisigExecutionInfo(transaction.executionInfo) ? transaction.executionInfo.nonce : undefined
-  const isReplacementTxPending = !isPending && nonce && pendingTxNonce && nonce === pendingTxNonce
+  const inBatch = batchTxActiveHover?.includes(transaction.id) ? ' highlight' : ''
```

Comment:
I like the idea, highlight is only used in this place but `will-be-replaced` is used in more than one place so it would be better to have a single source of truth. Lets create a new issue to address this refactor.